std/meta.zig

```diff
@@ -115,6 +115,66 @@ test "std.meta.Child" {
     testing.expect(Child(?u8) == u8);
 }
 
+/// Given an array/pointer type, return the slice type `[]Child`.
+/// Preserves all pointer attributes such as `const`/`volatile` etc.
+pub fn Slice(comptime T: type) type {
+    return switch (@typeInfo(T)) {
+        TypeId.Array => |info| @compileError("Slice not implemented for arrays"),
+        TypeId.Pointer => |info| @Type(builtin.TypeInfo { .Pointer = builtin.TypeInfo.Pointer {
+            .size = builtin.TypeInfo.Pointer.Size.Slice,
+            .is_const = info.is_const,
+            .is_volatile = info.is_volatile,
+            .alignment = info.alignment,
+            .child = info.child,
+            .is_allowzero = info.is_allowzero,
+        }}),
+        else => @compileError("Expected pointer or array type, " ++ "found '" ++ @typeName(T) ++ "'"),
+    };
+}
+
+test "std.meta.Slice" {
+    testing.expect(Slice([]u8) == []u8);
+    testing.expect(Slice([]const u8) == []const u8);
+    testing.expect(Slice(*u8) == []u8);
+    testing.expect(Slice(*const u8) == []const u8);
+    testing.expect(Slice([*]u8) == []u8);
+    testing.expect(Slice([*]const u8) == []const u8);
+    //testing.expect(Slice([10]u8) == []const u8);
```

Comment:
Let's have this either be a real test case or deleted. It's not clear what this comment is communicating here.

How about this test case?
`testing.expect(Slice(*[10]u8) == []const u8); `

Same comment for the other commented out test case below.

Reply:
I was thinking we would integrate https://github.com/ziglang/zig/pull/3173 first, then uncomment this.

Reply:
I think it's a smell that the function would accept a non-reference type and return a reference type. I think the test case I provided is the one you're looking for here.

Reply:
Oh sorry I misunderstood your point.  Yes I think `*[N]T` makes more sense than `[N]T`.  Will change

Reply:
So the question is, should `Slice(*[N]T)` become `[][N]T` or `[]T`?

Reply:
`Slice(*[N]T)` should become `[]T` to be consistent with how type coercion works in the rest of the language.

Reply:
Hmmm, given that answer, I'm not sure what the rules are.  It sounds like we have one or more special cases for single-value pointers?  What would these cases be?

```'
if (single_value_pointer)
    if (pointer_to_array) return []T.child.child
    else return []T.child
```

Are there any other cases where we would return `[]T.child.child`?

Reply:
> Are there any other cases where we would return []T.child.child?

Single item pointer to array is the only case that matches type coercion semantics